ast/src/parsed/display.rs

```diff
@@ -598,28 +598,37 @@ fn format_list<L: IntoIterator<Item = I>, I: Display>(list: L) -> String {
 impl<Ref: Display> Display for Expression<Ref> {
     fn fmt(&self, f: &mut Formatter<'_>) -> Result {
         match self {
-            Expression::Reference(reference) => write!(f, "{reference}"),
-            Expression::PublicReference(name) => write!(f, ":{name}"),
-            Expression::Number(Number { value, .. }) => write!(f, "{value}"),
-            Expression::String(value) => write!(f, "{}", quote(value)),
-            Expression::Tuple(items) => write!(f, "({})", format_list(items)),
-            Expression::LambdaExpression(lambda) => write!(f, "{lambda}"),
-            Expression::ArrayLiteral(array) => write!(f, "{array}"),
-            Expression::BinaryOperation(binaryop) => {
+            Expression::Reference(_, reference) => write!(f, "{reference}"),
+            Expression::PublicReference(_, name) => write!(f, ":{name}"),
+            Expression::Number(_, Number { value, .. }) => write!(f, "{value}"),
+            Expression::String(_, value) => write!(f, "{}", quote(value)),
+            Expression::Tuple(_, items) => write!(f, "({})", format_list(items)),
+            Expression::LambdaExpression(_, lambda) => write!(f, "{lambda}"),
+            Expression::ArrayLiteral(_, array) => write!(f, "{array}"),
+            Expression::BinaryOperation(_, binaryop) => {
                 write!(f, "{binaryop}")
             }
-            Expression::UnaryOperation(unaryop) => {
+            Expression::UnaryOperation(_, unaryop) => {
                 write!(f, "{unaryop}")
             }
-            Expression::IndexAccess(index_access) => write!(f, "{index_access}"),
-            Expression::FunctionCall(fun_call) => write!(f, "{fun_call}"),
-            Expression::FreeInput(input) => write!(f, "${{ {input} }}"),
-            Expression::MatchExpression(match_expr) => {
-                write!(f, "{match_expr}")
+            Expression::IndexAccess(_, index_access) => write!(f, "{index_access}"),
+            Expression::FunctionCall(_, fun_call) => write!(f, "{fun_call}"),
+            Expression::FreeInput(_, input) => write!(f, "${{ {input} }}"),
+            Expression::MatchExpression(_, MatchExpression { scrutinee, arms }) => {
+                writeln!(f, "match {scrutinee} {{")?;
+                write_items_indented(f, arms)?;
+                write!(f, "}}")
             }
-            Expression::IfExpression(e) => write!(f, "{e}"),
-            Expression::BlockExpression(block_expr) => {
-                write!(f, "{block_expr}")
+            Expression::IfExpression(_, e) => write!(f, "{e}"),
+            Expression::BlockExpression(_, BlockExpression { statements, expr }) => {
+                if statements.is_empty() {
```

Comment:
Looks like an old version as well.

Reply:
Oh, sorry, I must have forgotten some branch of the match during the merge, how embarrassing. It won't happen again :sweat: